cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -1058,7 +1058,16 @@ IceInternal::OutgoingConnectionFactory::ConnectCallback::nextConnector()
                     << _iter->connector->toString();
             }
             Ice::ConnectionIPtr connection = _factory->createConnection(_iter->connector->connect(), *_iter);
-            connection->start(shared_from_this());
+            auto self = shared_from_this();
+            connection->start(
+                [self](ConnectionIPtr conn)
```

Comment:
I'd rather call the connection "connection" unless it triggers some name hidding warning

Reply:
Yes it does:

```
src/Ice/ConnectionFactory.cpp:1063:39: error: declaration shadows a local variable [-Werror,-Wshadow-uncaptured-local]
                [self](ConnectionIPtr connection)
                                      ^
src/Ice/ConnectionFactory.cpp:1060:33: note: previous declaration is here
            Ice::ConnectionIPtr connection = _factory->createConnection(_iter->connector->connect(), *_iter);
```